src/MyPlot/provider/MySQLProvider.php

```diff
@@ -219,22 +252,24 @@ private function reconnect() : bool {
 					$level->save(); // don't force in case owner doesn't want it saved
 					$level->unload(true); // force unload to prevent possible griefing
 				}
-				//$this->plugin->getServer()->shutdown(); // TODO: create config setting for this
+				if($this->plugin->getConfig()->getNested("MySQLSettings.ShutdownOnFailure", false)) {
+					$this->plugin->getServer()->shutdown();
+				}
 				return false;
 			}
 		}
 		return true;
 	}
-
-	private function prepare(){
+	
+	private function prepare() : void {
 		$this->sqlGetPlot = $this->db->prepare(
-			"SELECT id, name, owner, helpers, denied, biome FROM plots WHERE level = ? AND X = ? AND Z = ?;"
+			"SELECT id, name, owner, helpers, denied, biome, done FROM plots WHERE level = ? AND X = ? AND Z = ?;"
 		);
 		$this->sqlSavePlot = $this->db->prepare(
-			"INSERT INTO plots (`id`, `level`, `X`, `Z`, `name`, `owner`, `helpers`, `denied`, `biome`) VALUES((SELECT id	FROM plots p WHERE p.level = ? AND X = ? AND Z = ?),?,?,?,?,?,?,?,?) ON DUPLICATE KEY UPDATE name = VALUES(name), owner = VALUES(owner), helpers = VALUES(helpers), denied = VALUES(denied), biome = VALUES(biome);"
+			"INSERT INTO plots (`id`, `level`, `X`, `Z`, `name`, `owner`, `helpers`, `denied`, `biome`, `done`) VALUES((SELECT id	FROM plots p WHERE p.level = ? AND X = ? AND Z = ?),?,?,?,?,?,?,?,?,?) ON DUPLICATE KEY UPDATE name = VALUES(name), owner = VALUES(owner), helpers = VALUES(helpers), denied = VALUES(denied), biome = VALUES(biome), done = VALUES(done);"
 		);
 		$this->sqlSavePlotById = $this->db->prepare(
-			"UPDATE plots SET id = ?, level = ?, X = ?, Z = ?, name = ?, owner = ?, helpers = ?, denied = ?, biome = ? WHERE id = VALUES(id);"
+			"UPDATE plots SET id = ?, level = ?, X = ?, Z = ?, name = ?, owner = ?, helpers = ?, denied = ?, biome = ?, done = ? WHERE id = VALUES(id);"
```

Comment:
This might have to be changed in some way. VALUES(xyz) sometimes does not work and disconnects from the database instead - depending on the input. So far, i have not seen the issue i MyPlot, but an UserManager plugin i am writing, where i tried to use VALUES in a SELECT.

Reply:
I'll leave MySQL to you. I don't know the language yet :man_shrugging: